src/app/OperationalSessionSetup.cpp

```diff
@@ -93,6 +93,13 @@ void OperationalSessionSetup::Connect(Callback::Callback<OnDeviceConnected> * on
     //
     EnqueueConnectionCallbacks(onConnection, onFailure);
 
+    if (mPerformingLookupOnConnectedSession)
+    {
+        // If there is an ongoing address lookup happening, now that we have enqueued the callbacks
+        // we can exit early. Callbacks will be called upon completino of address resolution.
```

Comment:
I don't understand how this works.

Let's say we have a session A.  We start doing a lookup for its peer, so create an `OperationalSessionSetup`.

While that lookup is happening, session A gets marked defunct and we try to create a new session to that same peer.  We end up in this code, find this existing OperationalSessionSetup, see mPerformingLookupOnConnectedSession is true and return here.

Then what happens?  What causes us to do CASE for the new session setup?

Reply:
Fixed using the recommendation below